packages/lib/src/vis/matrix/StickyGrid.tsx

```diff
@@ -1,39 +1,58 @@
-import { useContext, useState } from 'react';
-import type { ReactNode, Ref } from 'react';
+import type { ReactNode } from 'react';
+import { useContext, forwardRef } from 'react';
 
-import AnchorCell from './AnchorCell';
-import { GridSettingsContext } from './GridSettingsContext';
-import IndexTrack from './IndexTrack';
+import HeaderCells from './HeaderCells';
 import styles from './MatrixVis.module.css';
+import { SettingsContext, RenderedItemsContext } from './context';
 
 interface Props {
   children: ReactNode;
   style: React.CSSProperties;
 }
 
-function StickyGrid(props: Props, ref: Ref<HTMLDivElement>) {
+const StickyGrid = forwardRef<HTMLDivElement, Props>((props, ref) => {
   const { children, style } = props;
-  const { rowCount, columnCount } = useContext(GridSettingsContext);
+  const { rowCount, columnCount, cellSize, sticky } =
+    useContext(SettingsContext);
 
-  const [sticky, setSticky] = useState(true);
+  const renderedItems = useContext(RenderedItemsContext);
+  const {
+    overscanColumnStartIndex: colStart = 0,
+    overscanColumnStopIndex: colStop = 0,
+    overscanRowStartIndex: rowStart = 0,
+    overscanRowStopIndex: rowStop = 0,
+  } = renderedItems || {};
 
   return (
     <div
-      className={styles.stickyGrid}
       ref={ref}
+      className={styles.stickyGrid}
       style={style}
       role="table"
+      aria-rowcount={rowCount}
+      aria-colcount={columnCount}
       data-sticky={sticky || undefined}
     >
-      <IndexTrack className={styles.indexRow} cellCount={columnCount - 1}>
-        <AnchorCell sticky={sticky} onToggle={() => setSticky(!sticky)} />
-      </IndexTrack>
+      <div className={styles.colHeaders}>
+        <div className={styles.topLeftCell} style={cellSize} data-bg />
+        <HeaderCells
+          indexMin={colStart}
+          indexMax={colStop}
+          transform={`translateX(${cellSize.width * colStart}px)`}
```

Comment:
This is the CSS `transform` I mentioned earlier. Since we're not rendering header cells from `0` to `colStart`, we have to shift the ones that are rendered by the amount of space the unrendered cells would have taken. I've left a component explaining this against the `transform` prop in `HeaderCells`.